File: 例子/lua的例子/EncryptedDataEncrypt.lua
local CryptCore=require("CryptCore")

function ToANSIString(utf8String)
    return CryptCore.UTF8ToACP(utf8String)
end



function Usage(filename)
    print(string.format("Usage: lua %s aes128|sm4 sha256|sm3 saltLen iterCount pwd inFileName outFileName [q7]",filename))
    print(ToANSIString("参数"))
    print(ToANSIString("\taes128|sm4：加密算法"))
    print(ToANSIString("\tsha256|sm3：Hash算法"))
	print(ToANSIString("\tsaltLen：盐值长度"))
	print(ToANSIString("\titerCount：迭代次数"))
	print(ToANSIString("\tpwd：口令"))
	print(ToANSIString("\tinFileName：明文文件名"))
	print(ToANSIString("\toutFileName：密文文件名"))
	print(ToANSIString("\tq7：可选，如果存在表示使用Q7格式"))
    print(ToANSIString("说明"))
    print(ToANSIString("\t使用pwd加密inFileName得到EncrypedData格式的数据保存在outFileName中"))
end

function GetFileConetent(fileName)
    local fIn=io.open(fileName,"rb")
    if fIn==nil then
        return nil
    end
    local data=fIn:read("*a")
    fIn:close()
    
    return data
end

function getHashAlgo(hashAlgo)
    hashAlgo=hashAlgo:upper()
    if hashAlgo== "SHA-1" or hashAlgo=="SHA1" then
        return CryptCore.Const.SHA1
    elseif  hashAlgo== "SM3" then
        return CryptCore.Const.SM3
    elseif  hashAlgo== "MD5" then
        return CryptCore.Const.MD5
    elseif hashAlgo== "SHA-224" or hashAlgo=="SHA224" then
        return CryptCore.Const.SHA224
    elseif hashAlgo== "SHA-256" or hashAlgo=="SHA256" then
        return CryptCore.Const.SHA256
    elseif hashAlgo== "SHA-384" or hashAlgo=="SHA384" then
        return CryptCore.Const.SHA384
    elseif hashAlgo== "SHA-512" or hashAlgo=="SHA512" then
        return CryptCore.Const.SHA512
    elseif hashAlgo== "SHA-512/224" then
        return CryptCore.Const["SHA-512/224"]
    elseif hashAlgo== "SHA-512/256" then
        return CryptCore.Const["SHA-512/256"]
    elseif hashAlgo== "SHA3-224" then
        return CryptCore.Const["SHA3-224"]
    elseif hashAlgo== "SHA3-256" then
        return CryptCore.Const["SHA3-256"]
    elseif hashAlgo== "SHA3-384" then
        return CryptCore.Const["SHA3-384"]
    elseif hashAlgo== "SHA3-512" then
        return CryptCore.Const["SHA3-512"]
    else
        return nil;
    end
end

function getEncAlgo(encAlgo)
    encAlgo=encAlgo:upper()
    if encAlgo== "AES-128" or encAlgo=="AES128" then
        return 2
    elseif  encAlgo== "AES-192" or encAlgo=="AES192" then
        return 3
    elseif  encAlgo== "AES-256" or encAlgo=="AES256" then
        return 4
    elseif encAlgo== "3DES" or encAlgo=="TDES" then
        return 1
    elseif encAlgo== "SM4" then
        return 5
    elseif encAlgo== "SM1" then
        return 6
    elseif encAlgo== "SSF33" then
        return 7
    else
        return nil;
    end
end

function save(outFileName,data)
	local fOut=assert(io.open(outFileName,"wb"))
	fOut:write(data)
	fOut:close()
end

if #arg~=7 and #arg~=8 then
    Usage(arg[0])
    os.exit(-1)
end

local encAlgo=getEncAlgo(arg[1])
if encAlgo==nil then
	print(ToANSIString("不支持的加密算法"))
	os.exit(-1)
end
local hashAlgo=getHashAlgo(arg[2])
if hashAlgo==nil then
	print(ToANSIString("不支持的Hash算法"))
	os.exit(-1)
end
local saltLen=tonumber(arg[3])
local iterCount=tonumber(arg[4])
local pwd=arg[5]
local inFileName=arg[6]
local outFileName=arg[7]
local flag=0
if #arg==8 then
	if arg[8]=="q7" then
		flag=1
	else
		Usage(arg[0])
		os.exit(-1)
	end
end

local clear=GetFileConetent(inFileName)
if clear==nil then
    print(ToANSIString("读明文文件失败"));
    os.exit(-1)
end

local encrypedData=CryptCore.EncryptedDataEncrypt(flag,encAlgo,hashAlgo,saltLen,iterCount,pwd,clear)
if encrypedData==nil then
	print(ToANSIString("加密失败"));
    os.exit(-1)
end
save(outFileName,encrypedData)
print(ToANSIString("加密成功"));

File: 例子/Java的例子/pure/EncryptedDataEncrypt.java
import net.netca.pki.encoding.asn1.pki.cms.EncryptedData;
import net.netca.pki.encoding.asn1.pki.cms.EncryptedDataBuilder;
import net.netca.pki.encoding.asn1.pki.Pkcs5PBKDF2;
import net.netca.pki.encoding.asn1.pki.NetcaSecureRandomGenerator;
import net.netca.pki.encoding.asn1.pki.NetcaSymEncrypter;
import net.netca.pki.encoding.asn1.pki.NetcaHasher;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;


public class EncryptedDataEncrypt {

	public static void main(String[] args) throws Exception{
		if(args.length!=5) {
			usage();
            return;
        }
		
		String pwd=args[0];
		String encAlgoName=args[1];
		String hashAlgoName=args[2];
		String inFileName=args[3];
		String outFileName=args[4];
		
		int encAlgo=getEncAlgo(encAlgoName);
		int hashAlgo=getHashAlgo(hashAlgoName);
		byte[] clear=readFile(inFileName);
		
		Pkcs5PBKDF2 kdf=new Pkcs5PBKDF2(new NetcaHasher());
		EncryptedData encryptedData=EncryptedDataBuilder.getInstance()
			.setEncryptAlgorithm(encAlgo)
			.setPkcs5KDFImplement(kdf, hashAlgo, 64, 8192)
			.setPwd(pwd)
			.setSymEncrypter(new NetcaSymEncrypter())
			.setSecureRandomGenerator(new NetcaSecureRandomGenerator())
			.encypt(clear);
		byte[] cipher=encryptedData.encode(true);
		writeFile(outFileName,cipher);
		System.out.println("Success");
	}
	
	private static void usage() {
		System.out.println("Usage: java EncryptedDataEncrypt pwd aes-128|sm4 sha256|sm3 clearFile cipherFile");
	}
	
	private static int getEncAlgo(String name) throws Exception{
		String upperName=name.toUpperCase();
		if(upperName.equals("AES-128")
				|| upperName.equals("AES128")) {
			return EncryptedDataBuilder.AES_128;
		} else if(upperName.equals("AES-192")
				|| upperName.equals("AES192")) {
			return EncryptedDataBuilder.AES_192;
		} else if(upperName.equals("AES-256")
				|| upperName.equals("AES256")) {
			return EncryptedDataBuilder.AES_256;
		} /*else if(upperName.equals("SM1")) {
			return EncryptedDataBuilder.SM1;
		} else if(upperName.equals("SSF33")) {
			return EncryptedDataBuilder.SSF33;
		}*/ else if(upperName.equals("SM4")) {
			return EncryptedDataBuilder.SM4;
		} else {
			throw new Exception("不支持的加密算法:"+name);
		}
	}
	private static int getHashAlgo(String name) throws Exception{
		String upperName=name.toUpperCase();
		if(upperName.equals("SHA-1")
				|| upperName.equals("SHA1")) {
			return EncryptedDataBuilder.SHA1;
		} else if(upperName.equals("SHA-224")
				|| upperName.equals("SHA224")) {
			return EncryptedDataBuilder.SHA224;
		} else if(upperName.equals("SHA-256")
				|| upperName.equals("SHA256")) {
			return EncryptedDataBuilder.SHA256;
		} else if(upperName.equals("SHA-384")
				|| upperName.equals("SHA384")) {
			return EncryptedDataBuilder.SHA384;
		} else if(upperName.equals("SHA-512")
				|| upperName.equals("SHA512")) {
			return EncryptedDataBuilder.SHA512;
		} else if(upperName.equals("SHA-512/224")
				|| upperName.equals("SHA512/224")) {
			return EncryptedDataBuilder.SHA512_224;
		} else if(upperName.equals("SHA-512/256")
				|| upperName.equals("SHA512/256")) {
			return EncryptedDataBuilder.SHA512_256;
		} else if(upperName.equals("SHA3-224")) {
			return EncryptedDataBuilder.SHA3_224;
		} else if(upperName.equals("SHA3-256")) {
			return EncryptedDataBuilder.SHA3_256;
		} else if(upperName.equals("SHA3-384")) {
			return EncryptedDataBuilder.SHA3_384;
		} else if(upperName.equals("SHA3-512")) {
			return EncryptedDataBuilder.SHA3_512;
		} else if(upperName.equals("SM3")) {
			return EncryptedDataBuilder.SM3;
		} else {
			throw new Exception("不支持的Hash算法:"+name);
		}
	}
	
	private static byte[] readFile(String fileName) throws Exception{
		File file=new File(fileName);
		
		long fileLen=file.length();
		if(fileLen>=Integer.MAX_VALUE) {
			throw new Exception("file too big");
		}
		int len=(int)fileLen;
		byte[] data=new byte[len];
		FileInputStream fIn=new FileInputStream(file);
		
		try{
			if (fIn.read(data)!=len) {
				throw new Exception("read file fail");
			}
			return data;
		} finally {
			fIn.close();
		}
	}
	
	private static void writeFile(String fileName,byte[] data) throws Exception{
		FileOutputStream fOut=new FileOutputStream(fileName);
		try {
			fOut.write(data);
		} finally {
			fOut.close();
		}
		
	}
}
